Signal/src/ViewControllers/ConversationView/ConversationViewModel.m

```diff
@@ -458,19 +459,24 @@ - (void)uiDatabaseDidUpdateExternally:(NSNotification *)notification
 
     OWSLogVerbose(@"");
 
-    if (self.delegate.shouldObserveDBModifications) {
-        // External database modifications can't be converted into incremental updates,
-        // so rebuild everything.  This is expensive and usually isn't necessary, but
-        // there's no alternative.
-        //
-        // We don't need to do this if we're not observing db modifications since we'll
-        // do it when we resume.
-        [self resetMappings];
+    if (!self.delegate.isObservingVMUpdates) {
+        return;
     }
+
+    // External database modifications can't be converted into incremental updates,
+    // so rebuild everything.  This is expensive and usually isn't necessary, but
+    // there's no alternative.
+    //
+    // We don't need to do this if we're not observing db modifications since we'll
+    // do it when we resume.
+    [self resetMappings];
 }
 
 - (void)uiDatabaseWillUpdate:(NSNotification *)notification
 {
+    if (!self.delegate.isObservingVMUpdates) {
+        return;
+    }
```

Comment:
As an optimization, we don't need to fire will/did update events to the delegate if the view isn't observing VM updates.